airflow/providers/microsoft/azure/hooks/data_lake.py

```diff
@@ -357,10 +357,10 @@ def get_conn(self) -> DataLakeServiceClient:  # type: ignore[override]
             credential = conn.password
         else:
             managed_identity_client_id = self._get_field(extra, "managed_identity_client_id")
-            workload_identity_tenant_id = self._get_field(extra, "workload_identity_tenant_id")
-            credential = AzureIdentityCredentialAdapter(
-                managed_identity_client_id=managed_identity_client_id,
-                workload_identity_tenant_id=workload_identity_tenant_id,
+            credential = (
```

Comment:
Does this credential chain change backward compatible?

Reply:
Instead of using `ManagedIdentityCredential`, I think we should try something like the following instead https://github.com/apache/airflow/blob/bf9bb72bb631c098bfee34a372aefb332c3ffd64/airflow/providers/microsoft/azure/hooks/data_factory.py#L215-L220

Reply:
We might also want to add tests like https://github.com/apache/airflow/blob/bf9bb72bb631c098bfee34a372aefb332c3ffd64/tests/providers/microsoft/azure/hooks/test_data_factory.py#L184C24-L184C57